### PR TITLE
Introduce refresh token process

### PR DESCRIPTION
**Pull request by MartinPankraz**
Added requests for refresh token, error handling and stronger JWT validation

### CODE REVIEW

**Comment by vladvino** [comment]
@VitaliyKurokhtin please review when you have a minute.

**Review by VitaliyKurokhtin — CHANGES_REQUESTED** [review]

File: examples/Get X-CSRF token from SAP gateway using send request.policy.xml [hunk 1]
```diff
@@ -1,45 +1,47 @@
...
+              <set-method>HEAD</set-method>
+              <set-header name="X-CSRF-Token" exists-action="override">
+                  <value>Fetch</value>
+              </set-header>
+              <set-header name="Authorization" exists-action="override">
+                  <!-- forward authorization for csrf token fetch -->
+                  <value>@(context.Request.Headers.GetValueOrDefault("Authorization"))</value>
+              </set-header>
+              <!--  Don't expose APIM subscription key to the backend. -->
+              <set-header name="Ocp-Apim-Subscription-Key" exists-action="delete" />
```
[the remark this answers is not in the document]
Comment (MartinPankraz) [reply examples/Get X-CSRF token from SAP gateway using send request.policy.xml:24]: agreed

File: examples/Get X-CSRF token from SAP gateway using send request.policy.xml [hunk 2]
```diff
@@ -1,45 +1,47 @@
...
+          </send-request>
+          <!-- Extract the token and cookie from the "SAPCSRFToken" and set as header in the POST request. -->
+          <set-header name="X-CSRF-Token" exists-action="override">
+              <value>@(((IResponse)context.Variables["SAPCSRFToken"]).Headers.GetValueOrDefault("x-csrf-token"))</value>
```
[the remark this answers is not in the document]
Comment (MartinPankraz) [reply examples/Get X-CSRF token from SAP gateway using send request.policy.xml]: done

**Review by MartinPankraz — COMMENTED** [review]
let me know if anything else

**Review by VitaliyKurokhtin — APPROVED** [review]

**Review by VitaliyKurokhtin — APPROVED** [review]

**Review by VitaliyKurokhtin — COMMENTED** [review]
all ok

**Pull request merged**